src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -211,6 +236,262 @@ public function deleteAction(Request $request, int $productId): Response
+    }
+
+    /**
+     * Delete products in bulk action.
+     *
+     * @AdminSecurity(
+     *     "is_granted('delete', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_products_v2_index",
+     *     message="You do not have permission to delete this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkDeleteAction(Request $request): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(new BulkDeleteProductCommand(
+                    $this->getProductIdsFromRequest($request))
+            );
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion', 'Admin.Notifications.Success')
+            );
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_products_v2_index');
+    }
+
+    /**
+     * Enable products in bulk action.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_products_v2_index",
+     *     message="You do not have permission to edit this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkEnableAction(Request $request): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(
+                new BulkToggleProductCommand(
+                    $this->getProductIdsFromRequest($request),
+                    true
+                )
+            );
+            $this->addFlash(
+                'success',
+                $this->trans('Products successfully activated.', 'Admin.Catalog.Notification')
+            );
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_products_v2_index');
+    }
+
+    /**
+     * Disable products in bulk action.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_products_v2_index",
+     *     message="You do not have permission to edit this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkDisableAction(Request $request): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(
+                new BulkToggleProductCommand(
+                    $this->getProductIdsFromRequest($request),
+                    false
+                )
+            );
+            $this->addFlash(
+                'success',
+                $this->trans('Products successfully deactivated.', 'Admin.Catalog.Notification')
```

Comment:
Well, not for sure, but in all likehood user will user other button to disable single products. This is bulk disabling, so I would guess it would be user to disable multiple products majority of the time.

Reply:
It's true that the user may usually rely on the row button and enable/disable products one by one However the bulk edition also makes sense (let's imagine you want to enable all the products of a category at once because they are seasonal ones like Christmas or Halloween decorations)

In this case, since the action is about several products a confirmation about the number of modified products is always nice. But it also depends on how we want to handle the bulk actions (see previous comments):
- one single bulk request that may fail because it's too long to too memory consuming
- several single requests displayed in a progression modal

Reply:
Yep, I was just asking if this wording always applies to the deletion of several products. As @JevgenijVisockij said, if this is a bulk action, then the plural form is ok. :-)